web3/contract.py

```diff
@@ -1085,7 +1085,7 @@ def _parse_logs(self, txn_receipt: TxReceipt, errors: EventLogErrorFlags) -> Ite
                 elif errors == IGNORE:
                     new_log = MutableAttributeDict(log)
                     new_log['errors'] = e
-                    rich_log = AttributeDict(new_log)
+                    rich_log = AttributeDict(cast(Dict[Any, Any], new_log))
```

Comment:
seems like the type here could be tightened since we know roughly what a log looks like...

Reply:
If you can't tighten up these `Any` types, maybe leave a comment indicating why?